master/internal/api_experiment.go

```diff
@@ -352,70 +350,128 @@ func (a *apiServer) DeleteExperiment(
 func (a *apiServer) DeleteExperiments(
 	ctx context.Context, req *apiv1.DeleteExperimentsRequest,
 ) (*apiv1.DeleteExperimentsResponse, error) {
-	_, _, err := grpcutil.GetUser(ctx)
+	curUser, _, err := grpcutil.GetUser(ctx)
 	if err != nil {
-		return nil, err
+		return nil, status.Errorf(codes.Internal, "failed to get the user: %s", err)
 	}
 
-	return &apiv1.DeleteExperimentsResponse{}, err
-}
+	results, experiments, err := exputil.DeleteExperiments(ctx, a.m.system, req.ExperimentIds,
+		req.Filters)
 
-func (a *apiServer) deleteExperiment(exp *model.Experiment, userModel *model.User) error {
-	agentUserGroup, err := user.GetAgentUserGroup(*exp.OwnerID, exp)
-	if err != nil {
-		return err
-	}
+	go func() {
```

Comment:
wish we had a meaningful `context.Context` here.. would be a really nice use case for a 'graceful shutdown of master' type of use case. (not on your code, just saying)